api/test/integration/conftest.py

```diff
@@ -67,56 +87,146 @@ def get_token_login_api():
+                                variables['request'].session.items))[-1]
+        test_filename = os.path.basename(variables['request'].path).split('_')
+        setup_environment(test_filename)
     urllib3.disable_warnings()
     variables["test_login_token"] = get_token_login_api()
+    
+    
+def pytest_tavern_beta_after_every_test_run(test_dict: dict, variables: dict):
+    """Pytest tavern hook function.
+       Cleans up the environment on the last test of each file
+
+        Parameters
+        ----------
+        test_dict : dict
+            test dictionary with test information.
+        variables : dict
+            dictionary with pytest related variables.
+    """
+    global last_test
+    if last_test.name == test_dict['test_name']:
+        last_test = None
+        global last_testsfailed
+        testsfailed = variables['request'].session.testsfailed - last_testsfailed
+        last_testsfailed = variables['request'].session.testsfailed
+        test_filename = os.path.basename(variables['request'].path).split('_')
+        clean_up_env(test_filename, testsfailed)
+
+def setup_environment(test_filename: list):
+    """Prepares the environment based on rbac white or black mode.
+
+        Parameters
+        ----------
+        test_filename : list
+            test filename splitted in a list of strings
+    """
+    if 'rbac' in test_filename:
+        rbac_mode = test_filename[2]
+        module = test_filename[3]
+    else:
+        rbac_mode = None
+        module = test_filename[1]
 
+    clean_tmp_folder()
 
-def build_and_up(env_mode: str, interval: int = 10, interval_build_env: int = 10,
-                 build: bool = True) -> dict:
-    """Build all Docker environments needed for the current test.
+    if rbac_mode:
+        change_rbac_mode(rbac_mode)
+        rbac_custom_config_generator(module, rbac_mode)
+    else:
+        enable_white_mode()
 
-    Parameters
-    ----------
-    env_mode : str
-        Indicates the environment to be used in the process.
-    interval : int
-        Time interval between every healthcheck.
-    interval_build_env : int
-        Time interval between every docker environment healthcheck.
-    build : bool
-        Flag to indicate if images need to be built.
+    general_procedure(module)
+    start_containers()
+    wait_env_up()
+
+def clean_up_env(test_filename: list, testsfailed: int):
+    """Cleans temporary folder, save environment logs and status,
+       stops and removes all Docker containers.
+
+        Parameters
+        ----------
+        test_filename : list
+            test filename splitted in a list of strings
+        testfailed: int
+            number of tests failed.
+    """
+    if 'rbac' in test_filename:
+        rbac_mode = test_filename[2]
+        module = test_filename[3]
+    else:
+        rbac_mode = None
+        module = test_filename[1]
```

Comment:
These lines are repeated above. Please put it into a function to reuse the code.